chips/nrf52/src/spi.rs

```diff
@@ -413,14 +424,19 @@ impl hil::spi::SpiMaster for SPIM {
     // Returns the actual rate set
     fn set_rate(&self, rate: u32) -> u32 {
```

Comment:
hmm, I don't like this `set_rate` and `get_rate` API. Do you know the reason why we cant have a more typed API such as:

- `fn set_rate(&self, freq: Frequency)`
- `fn get_rate(&self) -> Frequency`

?

Reply:
It's not great, but I don't think we can do much better easily. For instance, the SiFive SPI implementation allows setting arbitrary SPI clocks as the register value is just a divider of the bus frequency. Ultimately, having an enum of all allowed frequency values would get very large very quickly - and will set a deviating frequency on boards where the requested one isn't available (so no real benefit here).

Reply:
Ok, I see. We could do is to make `Frequency` an associated type and each impl has to decide how to construct it but it is out-of-scope for this PR.

Reply:
Yes, I thought about that for a while. I'm not sure whether that is worth the additional effort for clients. We'd need to at least make it a `dyn Trait` which allows calling a `frequency_lower_eq()`, returning a Frequency variant being lower or equals to the requested one. Otherwise HW drivers will be unable to automatically set the required frequency agnostic of the underlying SPI implementation.